test/src/androidTest/kotlin/io/realm/LinkTests.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm
+
+import io.realm.runtimeapi.RealmModule
+import test.link.Child
+import test.link.Parent
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+
+class LinkTests {
+
+    @RealmModule(Parent::class, Child::class)
+    class MySchema
+
+    lateinit var realm: Realm
+
+    @BeforeTest
+    fun setup() {
+        val configuration = RealmConfiguration.Builder(schema = MySchema()).build()
+        realm = Realm.open(configuration)
+        // FIXME Cleaning up realm to overcome lack of support for deleting actual files
```

Comment:
Check my GC PR where I introduced the platform-specific routines to create and delete temporary directories for test isolation

Reply:
Added a TODO with link to it. Then we can see who gets to merge first.

Reply:
I update all the tests to use your new tmp-dir infrastructure. A small note is that the shared tests cannot use `@ ExperimentalPathApi` as it is only available for JVM. I added the opt-in as a compiler command line arguments instead.